### PR TITLE
Support TLS termination in falconer

### PR DESCRIPTION
**Pull request by sjung-stripe**
What it says on the tin.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -5,7 +5,14 @@ type Config struct {
...
 	StatsdAddress                 string `yaml:"statsd_address"`
-	WorkerCount                   int    `yaml:"worker_count"`
-	WorkerSpanDepth               int    `yaml:"worker_span_depth"`
-	WorkerWatchDepth              int    `yaml:"worker_watch_depth"`
+	TLS                           struct {
+		Certificates []struct {
+			Cert string `yaml:"cert"`
+			Key  string `yaml:"key"`
+		} `yaml:"certificates"`
+		ClientCas []string `yaml:"client_cas"`
```
Comment (sdboyer-stripe) [inline config.go]: nit: `ClientCAs` (does the generator even let you do that?)

**Comment by sjung-stripe** [comment]
changed CI configuration to use the new `dep check` instead of `dep ensure -v`

File: Dockerfile [hunk 2]
```diff
@@ -32,7 +32,7 @@ RUN git reset --hard HEAD && git status
 # because we are guaranteed only one version of Go
 # used to build protoc-gen-go
 RUN go generate
-RUN dep ensure -v
+RUN dep check
 # Exclude vendor from gofmt checks.
 RUN mv vendor ../ && gofmt -w . && mv ../vendor .
```
Comment (sdboyer-stripe) [inline Dockerfile]: i think there are some other tweaks we can make here, too, but i'll deal with them later.

File: cmd/falconer/main.go [hunk 3]
```diff
@@ -42,6 +46,31 @@ func main() {
...
+
+		for _, caConfig := range config.TLS.ClientCas {
+			caData, err := ioutil.ReadFile(caConfig)
+			if err != nil {
+				log.WithError(err).Fatal("Failed to read client CA")
+			}
+			if !tlsConfig.ClientCAs.AppendCertsFromPEM(caData) {
+				log.Fatal("Failed to load client CA")
+			}
+			tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
```
Comment (sdboyer-stripe) [inline cmd/falconer/main.go]: Seems like doing this unconditionally might be more than we want? Maybe we can add configurability around it later, if needed.

File: config.go (hunk 1 above)
Comment (sjung-stripe) [reply config.go]: lemme check, but i don't think so

File: cmd/falconer/main.go (hunk 3 above)
Comment (sjung-stripe) [reply cmd/falconer/main.go]: i'll add configuration for it

**Pull request merged**